src/main/java/com/mykaarma/urlshortener/model/AvailableHashPool.java

```diff
@@ -0,0 +1,19 @@
+package com.mykaarma.urlshortener.model;
+
+import java.io.Serializable;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+/**
+ * AvailableHashPool maintains a pool of hashes, which can be used by the url-shortener. isValid true indicates that the hashes are unused;
```

Comment:
@AadiS11 Maybe we should rename this field to isAvailable?

Reply:
Renamed to isAvailable